pkg/services/govmomi/service.go

```diff
@@ -71,15 +72,17 @@ func (vms *VMService) ReconcileVM(ctx *context.VMContext) (vm infrav1.VirtualMac
 	// Before going further, we need the VM's managed object reference.
 	vmRef, err := findVM(ctx)
 	if err != nil {
-		if !isNotFound(err) {
+		if !errors.As(err,find.NotFoundError{}) {
 			return vm, err
 		}
 
 		// If the machine was not found by BIOS UUID it means that it got deleted from vcenter directly
-		if wasNotFoundByBIOSUUID(err) {
-			ctx.VSphereVM.Status.FailureReason = capierrors.MachineStatusErrorPtr(capierrors.UpdateMachineError)
-			ctx.VSphereVM.Status.FailureMessage = pointer.StringPtr(fmt.Sprintf("Unable to find VM by BIOS UUID %s. The vm was removed from infra", ctx.VSphereVM.Spec.BiosUUID))
-			return vm, err
+		if errors.As(err,find.NotFoundError{}) {
```

Comment:
don't need this extra IF as if it gets past line 75, this condition would be true